internal/api/server/http_socket.go

```diff
@@ -14,9 +14,16 @@ import (
 
 var apiServer *http.Server
 
-const API_SERVER_STOP_LOCK = "internal.api.server.API_STOP_LOCK"
+const API_SERVER_LOCK = "internal.api.server.API_SERVER_LOCK"
 
 func runAPIServer(routes *[]core.Route) (err error) {
+	common.Lock(API_SERVER_LOCK)
+	if apiServer != nil {
+		common.Unlock(API_SERVER_LOCK)
+		return fmt.Errorf("API server is already running.")
+	}
+	common.Unlock(API_SERVER_LOCK)
```

Comment:
I know I asked for this to be first and all together, but I think we can make it a bit better.
I'm thinking:
 - Do the recover defer first in the function, followed by this.
 - Move the unlock until after you set apiServer (note that there are no possible errors or returns in that section.

This way there shouldn't be any holes where two people can call runServer().